pages/register.tsx

```diff
@@ -0,0 +1,290 @@
+import React, { ChangeEvent, useState } from 'react';
+import PageLayout from "../components/layout/PageLayout";
+import styles from "../styles/pages/register.module.scss";
+import TextField from '@mui/material/TextField';
+import InputLabel from '@mui/material/InputLabel';
+import MenuItem from '@mui/material/MenuItem';
+import FormControl from '@mui/material/FormControl';
+import Select, { SelectChangeEvent } from '@mui/material/Select';
+import Link from "next/link";
+import { BiErrorCircle } from "react-icons/bi"
+import ErrorTextBox from '../components/register/ErrorTextBox';
+
+function Register() {
+
+    const [formValues, setFormValues] = useState({
+        firstName: '',
+        lastName: '',
+        branch: branchNames.length === 0 ? '' : branchNames[0],
+        sid: '',
+        username: '',
+        email: '',
+        password: '',
+        confirmPassword: '',
+    })
+
+    const [error, setError] = useState(false);
+
+    const handleChange = (event: ChangeEvent<HTMLInputElement>
+        | SelectChangeEvent
+        | React.ChangeEvent<HTMLTextAreaElement
+            | HTMLInputElement>) => {
+        event.preventDefault();
+        const name = event.target.name;
+        const value = event.target.value;
+        setFormValues({
+            ...formValues,
+            [name]: value
+        })
+    }
+
+    const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+        event.preventDefault();
+
+        if (!isSidCorrect(formValues.sid) || !arePasswordsMatching(formValues.password, formValues.confirmPassword)) {
+            setError(true);
+
+            setTimeout(() => {
+                setError(false);
+            }, 3000);
```

Comment:
I would say create an ErrorHandler at global level, which will be a toast or a dialog and timeout is handled inside of that, rather than calling this in every component.

Something like

```js
if(error) {
   showDialog(errorTitle, errorDescription, time[optional]);
}
```

And the `showDialog` function will simply render a dialog on the whole screen no matter in which component the error is thrown.

Reply:
Building over a [toast library](https://www.npmjs.com/package/react-toastify) allowed?

Reply:
Ofc everything is allowed, just (make it look that we did it ourselves).

But if you can do a dialog, it ll be better, toasts are annoying